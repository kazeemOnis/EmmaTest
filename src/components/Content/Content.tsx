import React from 'react';
import {Dimensions} from 'react-native';
import styled from '../../theme';

interface ContentProps {
  title?: string;
  subtext?: string;
  content?: string;
}

const Wrapper = styled.View`
  flex: 1;
  height: ${Dimensions.get('window').height}px;
`;

const Title = styled.Text`
  font-size: 24px;
  font-family: 'Nunito'
  color: white;
  text-align: center;
  margin-bottom: 4px;
`;

const Subtext = styled.Text`
  font-family: 'Nunito';
  color: white;
  text-align: center;
  font-size: 16px;
`;

const ContentText = styled.Text`
  font-family: 'Nunito';
  color: white;
  font-size: 14px;
  margin-top: 16px;
`;

export const Content = ({title, subtext, content}: ContentProps) => {
  return (
    <Wrapper>
      {title && <Title>{title}</Title>}
      {subtext && <Subtext>{subtext}</Subtext>}
      {content && <ContentText>{content}</ContentText>}
    </Wrapper>
  );
};
