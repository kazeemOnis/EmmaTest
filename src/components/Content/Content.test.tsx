import React from 'react';
import {cleanup, render} from '@testing-library/react-native';
import {act} from 'react-test-renderer';
import {Content} from './Content';

afterEach(() => {
  cleanup();
});

const mockProps = {
  title: 'Kazeem',
  subtext: 'React Native Engineer',
  content:
    'Experienced React Native Engineer with over 4 years of experience in building and shipping production ready apps to users',
};

const setup = (overrides = {}) => {
  const props = {...mockProps, ...overrides};
  const utils = render(<Content {...props} />);
  return {...utils, props};
};

describe('Content', () => {
  it('Should render content', async () => {
    const {toJSON} = setup();
    await act(async () => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('Should render all text when title, subtext, content are given', () => {
    const {queryByText} = setup();

    expect(queryByText('Kazeem')).toBeTruthy();
    expect(queryByText('React Native Engineer')).toBeTruthy();
    expect(
      queryByText(
        'Experienced React Native Engineer with over 4 years of experience in building and shipping production ready apps to users',
      ),
    ).toBeTruthy();
  });

  it('Should not render text when title, subtext, content are not given', async () => {
    const {queryByTestId, toJSON} = setup({
      title: '',
      subtext: '',
      content: '',
    });
    await act(async () => {
      expect(toJSON()).toMatchSnapshot();
      expect(queryByTestId('title')).toBeNull();
      expect(queryByTestId('subtext')).toBeNull();
      expect(queryByTestId('content')).toBeNull();
    });
  });
});
