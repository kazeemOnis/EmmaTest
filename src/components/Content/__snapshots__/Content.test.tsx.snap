// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Should not render text when title, subtext, content are not given 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 1334,
      },
    ]
  }
/>
`;

exports[`Content Should render content 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 1334,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontFamily": "Nunito",
          "fontSize": 24,
          "marginBottom": 4,
          "textAlign": "center",
        },
      ]
    }
    testID="contentTitle"
  >
    Kazeem
  </Text>
  <Text
    style={
      [
        {
          "color": "white",
          "fontFamily": "Nunito",
          "fontSize": 16,
          "textAlign": "center",
        },
      ]
    }
    testID="contentSubtext"
  >
    React Native Engineer
  </Text>
  <Text
    style={
      [
        {
          "color": "white",
          "fontFamily": "Nunito",
          "fontSize": 14,
          "marginTop": 16,
        },
      ]
    }
    testID="contentContent"
  >
    Experienced React Native Engineer with over 4 years of experience in building and shipping production ready apps to users
  </Text>
</View>
`;
